runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -176,30 +177,53 @@ def model(self, id: str) -> Optional[prophet.forecaster.Prophet]:
             return model_from_json(self._model_json[id])
         return None
 
-    def _make_future_dataframe(self, id: str, horizon: int, include_history: bool = True) -> pd.DataFrame:
+    def make_future_dataframe(
+            self,
+            horizon: Optional[int] = None,
+            include_history: bool = True,
+            groups: List[Tuple] = None,
+    ) -> pd.DataFrame:
         """
-        Generate future dataframe for one model by calling the API from prophet
-        :param id: identity for the Prophet model
+        Generate dataframe with future timestamps for all valid identities
+        :param horizon: Int number of periods in the future
+        :param frequency: frequency of the history time series. It should be valid frequency
+            for pd.date_range, such as "D" or "M"
+        :param groups: the collection of group(s) to generate forecast predictions.
+            The group definiteions must be the key value
+        :return: pd.DataFrame that extends forward from history_last_date
+        """
+        horizon=horizon or self._horizon
+        if groups is not None:
+            model_keys = set(self._model_json.keys())
+            if not set(groups).issubset(model_keys):
+                raise ValueError(f"Invalid groups: {set(groups) - model_keys}.")
+        else:
+            groups = list(self._model_json.keys())
+
+        end_time = pd.Timestamp(self._timeseries_end)
+        future_df = make_future_dataframe(
+            start_time=self._timeseries_starts,
+            end_time=end_time,
+            horizon=horizon,
+            frequency=self._frequency,
+            include_history=include_history,
+            groups=groups,
+            group_names=self._id_cols
+        )
+        return future_df
+
+    def _make_future_dataframe(self, id: Tuple, horizon: int, include_history: bool = True) -> pd.DataFrame:
```

Comment:
Shall we change the method `_make_future_dataframe` in the univariate model wrapper `ProphetModel` as well to use our own util function for consistency? It's still calling self.model().make_future_dataframe.

Reply:
Good point. I will make a follow up on this. Currently `ProphetModel` does not contain time series starts and ends. So it is not simple to use this util function.